peerinst/views/collection.py

```diff
@@ -43,5 +65,240 @@ def get_form_kwargs(self):
+    def get_queryset(self):
+        teacher = get_object_or_404(Teacher, user=self.request.user)
+        return Collection.objects.filter(owner=teacher)
+
+
+class FollowedCollectionListView(
+    LoginRequiredMixin, NoStudentsMixin, ListView
+):
+    """
+    list for collections followed by teacher that are public
+    or owned by teacher
+    """
+
+    model = Collection
+    template_name = "peerinst/collection/followed_collection_list.html"
+
+    def get_queryset(self):
+        teacher = get_object_or_404(Teacher, user=self.request.user)
+        return Collection.objects.filter(
+            Q(followers=teacher, private=False)
+            | Q(followers=teacher, owner=teacher)
+        )
+
+
+class FeaturedCollectionListView(
+    LoginRequiredMixin, NoStudentsMixin, ListView
+):
+    """
+    list for featured collections that are public or owned by teacher
+    """
+
+    model = Collection
+    template_name = "peerinst/collection/featured_collection_list.html"
+
+    def get_queryset(self):
+        teacher = get_object_or_404(Teacher, user=self.request.user)
+        return Collection.objects.filter(
+            Q(featured=True, private=False) | Q(featured=True, owner=teacher)
+        )
+
+
+def featured_collections(request):
+    data = serializers.serialize(
+        "json", Collection.objects.filter(featured=True)
+    )
+    return JsonResponse(data, safe=False)
+
+
+class CollectionDistributeDetailView(
+    LoginRequiredMixin, NoStudentsMixin, DetailView
+):
+    """
+    view to assign/unassign a collection's assignments from a student group
+    """
+
+    model = Collection
+    template_name = "peerinst/collection/collection_distribute.html"
+
+    def dispatch(self, *args, **kwargs):
+        teacher = get_object_or_404(Teacher, user=self.request.user)
+        if (
+            teacher == self.get_object().owner
+            or self.get_object().private is False
+        ):
+            return super(CollectionDistributeDetailView, self).dispatch(
+                *args, **kwargs
+            )
+        else:
+            raise PermissionDenied
+
+    def get_context_data(self, **kwargs):
+        context = super(DetailView, self).get_context_data(**kwargs)
+        teacher = get_object_or_404(Teacher, user=self.request.user)
+        collection = self.get_object()
+        context["student_groups"] = teacher.current_groups.all()
+        context["collection_data"] = collection_data(collection=collection)
+        context["group_data"] = {}
+        for group in teacher.current_groups.all():
+            context["group_data"][group.pk] = True
+            for assignment in collection.assignments.all():
+                if not StudentGroupAssignment.objects.filter(
+                    group=group, assignment=assignment
+                ).exists():
+                    context["group_data"][group.pk] = False
+        return context
+
+
+@teacher_required
+@require_POST
+def collection_add_assignment(request, teacher):
```

Comment:
For new code, functions should be proper English such as `add_assignment_to_collection`. If namespacing is required, use the module name to do so when importing and using the function.